# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/Localization.webidl
# Commit: 595529cd906f
# Full Hash: 595529cd906f94cdb2bc12a7a40072db0c07198e
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-03 22:01:19
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 9f3e2963d925 -r 595529cd906f dom/webidl/Localization.webidl
--- a/dom/webidl/Localization.webidl	Tue Aug 03 05:52:03 2021 +0000
+++ b/dom/webidl/Localization.webidl	Tue Aug 03 05:52:03 2021 +0000
@@ -44,24 +44,6 @@
 };
 
 /**
- * A callback function which takes a list of localization resources
- * and produces an iterator over FluentBundle objects used for
- * localization with fallbacks.
- */
-callback GenerateBundles = Promise<any> (sequence<DOMString> aResourceIds);
-callback GenerateBundlesSync = any (sequence<DOMString> aResourceIds);
-
-/**
- * The structure provides custom methods for the Localization API that
- * will be used to generate the `FluentBundle` iterator.
- * This allows the consumer to overload the default Gecko generator.
- */
-dictionary BundleGenerator {
-  GenerateBundles generateBundles;
-  GenerateBundlesSync generateBundlesSync;
-};
-
-/**
  * Localization is an implementation of the Fluent Localization API.
  *
  * An instance of a Localization class stores a state of a mix
@@ -99,14 +81,14 @@
   /**
    * A method for adding resources to the localization context.
    */
-  void addResourceIds(sequence<DOMString> aResourceIds);
+  void addResourceIds(sequence<UTF8String> aResourceIds);
 
   /**
    * A method for removing resources from the localization context.
    *
    * Returns a new count of resources used by the context.
    */
-  unsigned long removeResourceIds(sequence<DOMString> aResourceIds);
+  unsigned long removeResourceIds(sequence<UTF8String> aResourceIds);
 
   /**
    * Formats a value of a localization message with a given id.